### PR TITLE
Proposal Voting Section

### PR DESCRIPTION
**Pull request by 1emu**
## Demo changes
- [X] Vote count on an option should count represented delegators and delegators who voted the option
- [X] Remove option colors
- [X] Do not show options as disabled, but an option that is already voted can't be voted again (not clickable)
- [X] If my delegate voted and I am overruling the vote, I should be able to vote the same option

## Choices
- [X] Add maxLength validation to poll choices

## ChoiceButtons
- [X] When overruling your vote you should see your delegate avatar on the choice picked by them
- [X] If your delegate voted, and you didn't: If you have delegators, then all the choices should be displayed
- [X] Add ellipsis for options that span more than 3 rows
- [X] Display tooltip with full text for options with more than 50 chars

## VotedChoice
- [X] If you didn't vote but your delegate did, it should display the delegate avatar
- [X] If you voted, and the delegate voted the same, it should display the delegate avatar
- [X] If you don't have delegators and your delegate voted, you should only see the voted choice

## Choice progress
- [X] Add ellipsis and tooltip

## Delegations Label
- [X] It should read *Your delegate* in all cases, instead of showing the delegate avatar && address
- [X] Update delegate and delegators label options
- [x] Replace 'individual' for 'account'

## VotedChoice Label
- [x] If your delegate voted and you have delegators, the label should not take into account the delegate vote for the text displayed (it should only reflect your vote)

## Footer
- [X] Make the footer a component
- [X] Display Voting with vp // Voted with vp depending on the user vote
- [X] Display Change Vote / Overrule & Cancel depending on the vote/voting/delegateVote
- [x] If not enough VP, it should display a warning and show a 'Get VP' link. The VP should take into account delegated VP!
- [x] If not signed in, it should not be visible
- [x] Take into account delegators VP

## Tests
- [X] Add tests for all the possible combinations of votes, delegate votes, and delegators votes (double check)
- [X] Refactor `getProposalVotingSectionConfig` method
- [x] Refactor frontend UI tests
- [ ] Test what happens if your delegators include your delegate

### CODE REVIEW

File: src/entities/Votes/DelegationsLabelBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import Time from 'decentraland-gatsby/dist/utils/date/Time'
+import { intersection } from 'lodash'
+import { Vote } from './types'
+import { DelegationsLabelProps } from './utils'
+
+export class DelegationsLabelBuilder {
```
Comment (ncomerci) [inline src/entities/Votes/DelegationsLabelBuilder.ts:6]: 🧙🏼‍♂️🔥

File: src/hooks/useVotingSectionTestData.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  accountDelegate: string | null
+  delegators: string[]
+  votes: Record<string, Vote> | null | undefined
+  choices: string[]
+  nextCase?: () => void
+  previousCase?: () => void
+}
+
+export default function useVotingSectionTestData(): TestData | null {
+  const testing = true // TODO: this can all be deleted after demo, or we can use a feature flag || env var
```
Comment (ncomerci) [inline src/hooks/useVotingSectionTestData.ts:18]: This

**Review by ncomerci — COMMENTED** [review]
Beautiful, a few minor comments!

File: src/entities/Votes/utils.ts [hunk 3]
```diff
@@ -145,25 +147,36 @@ export function calculateResultWinner(choices: string[], votes: Record<string, V
...
+  values?: any
+  delegate?: string
+  voteCount?: number
+  totalVotes?: number
+}
+
+export interface DelegationsLabelProps {
+  delegateLabel?: { id: string; values?: any }
+  delegatorsLabel?: { id: string; values?: any }
+}
```
Comment (andyesp) [inline src/entities/Votes/utils.ts:176]: You can move this to Votes/types.ts

File: src/intl/en.json [hunk 4]
```diff
@@ -52,6 +52,7 @@
       "choices_insufficient": "Insufficient options",
       "choices_empty": "Some option is empty",
       "choices_too_short": "Some option is to short",
+      "choices_too_long": "Some option is to long",
```
Comment (andyesp) [inline src/intl/en.json:55]: Typo. `too`

File: src/components/Section/ProposalVotingSectionTestCases.tsx [hunk 5]
```diff
@@ -0,0 +1,190 @@
+import { TestData } from '../../hooks/useVotingSectionTestData'
+
```
Comment (andyesp) [inline src/components/Section/ProposalVotingSectionTestCases.tsx:2]: This file (ProposalVotingSectionTestCases.tsx) can be .ts as it is not returning tsx.

**Review by andyesp — APPROVED** [review]

**Review by andyesp — APPROVED** [review]

**Pull request merged**